include/mata/nfa/nfa.hh

```diff
@@ -206,8 +206,25 @@ public:
      */
     Nfa& trim(StateRenaming* state_renaming = nullptr);
 
+    /**
+     * @brief Returns vector ret where ret[q] is the length of the shortest path from any initial state to q
+     */
     std::vector<State> distances_from_initial() const;
```

Comment:
It would be nice if we have some general BFS/DFS method parameterized by callbacks (in a similar fashion as for Tarjan). All these functions such as `distances_from_initial` or `distances_to_final` could be easily expressible by this and you wouldn't need to write BFS/DFS all over again.

Reply:
Create an issue from this:  #446.